### DISCUSSION TITLE
open-webui@0.2.0 build  ModuleNotFoundError: No module named 'micropip'

### DISCUSSION THREAD

**Discussion by tqangxl** [open-webui/open-webui#2730] [General]
# Bug Report

## Description

PS D:\Lib\Dev\AI\open-webui\open-webui> npm run build

> open-webui@0.2.0 build
> npm run pyodide:fetch && vite build

> open-webui@0.2.0 pyodide:fetch
> node scripts/prepare-pyodide.js

Setting up pyodide + micropip

if(shouldRunNow)callMain(args);postRun()}if(Module["setStatus"]){Module["setStatus"]("Running...");setTimeout(function(){setTimeout(function(){Module["setStatus"]("")},1);doRun()},1)}else{doRun()}}if(Module["preInit"]){if(typeof Module["preInit"]=="function")Module["preInit"]=[Module["preInit"]];while(Module["preInit"].length>0){Module["preInit"].pop()()}}var shouldRunNow=true;if(Module["noInitialRun"])shouldRunNow=false;run();moduleRtn=readyPromise;

PythonError: Traceback (most recent call last):
  File "/lib/python312.zip/_pyodide/_base.py", line 629, in pyimport_impl
    res = __import__(stem, fromlist=fromlist)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'micropip'
The module 'micropip' is included in the Pyodide distribution, but it is not installed.
You can install it by calling:
  await micropip.install("micropip") in Python, or
  await pyodide.loadPackage("micropip") in JavaScript
See https://pyodide.org/en/stable/usage/loading-packages.html for more details.

    at new_error (D:\Lib\Dev\AI\open-webui\open-webui\node_modules\pyodide\pyodide.asm.js:10:9965)
    at wasm://wasm/0267d6e2:wasm-function[302]:0x16dbeb
    at wasm://wasm/0267d6e2:wasm-function[303]:0x16dcb1
    at Module._pythonexc2js (D:\Lib\Dev\AI\open-webui\open-webui\node_modules\pyodide\pyodide.asm.js:10:700519)
    at callPyObjectKwargs (D:\Lib\Dev\AI\open-webui\open-webui\node_modules\pyodide\pyodide.asm.js:10:64179)
    at Module.callPyObject (D:\Lib\Dev\AI\open-webui\open-webui\node_modules\pyodide\pyodide.asm.js:10:65369)
    at g.apply (D:\Lib\Dev\AI\open-webui\open-webui\node_modules\pyodide\pyodide.asm.js:10:80884)
    at Object.apply (D:\Lib\Dev\AI\open-webui\open-webui\node_modules\pyodide\pyodide.asm.js:10:78180)
    at Object.pyimport (D:\Lib\Dev\AI\open-webui\open-webui\node_modules\pyodide\pyodide.asm.js:10:108115)
    at downloadPackages (file:///D:/Lib/Dev/AI/open-webui/open-webui/scripts/prepare-pyodide.js:22:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///D:/Lib/Dev/AI/open-webui/open-webui/scripts/prepare-pyodide.js:38:1 {
  type: 'ModuleNotFoundError',
  __error_address: 16927776
}

Node.js v20.13.1
**Bug Summary:**

**Comment by Belonger** [comment]
yes，i also meet this bug，how to slove?

**Reply by Belonger** [reply]
i solved this bug，you need to install a new python，like python=3.11，python3.11 version also updating!

**Reply by tqangxl** [reply]
我的是Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)] on win32

**Reply by Belonger** [reply]
yes，i am also to, you need to create a new conda env, like: conda create -n new_env  python=3.11，and is ok!

**Reply by tqangxl** [reply]
update to v0.2.2 fixed
  9d4c07b7..b4fca046  dev        -> origin/dev
 * [new tag]           v0.2.0     -> v0.2.0
 * [new tag]           v0.2.1     -> v0.2.1
 * [new tag]           v0.2.2     -> v0.2.2

**Reply by tqangxl** [reply]
@Belonger thanks your replay!

**Comment by xunx911** [comment]
i meet both in 0.3.5 and 0.2.2 version,wsl ubuntu python3.11,creating a new conda env is not useful

**Reply by cheebinc** [reply]
Hi, is there any existing solution? I'm getting the same issue in v0.3.35 using python3.11 venv getting same error

**Comment by yusufix** [comment]
from pypika.dialects import (                                                                                                                                                                                                     │
│       ClickHouseQuery,                                                                                                                                                                                                              │
│       Dialects,                                                                                                                                                                                                                     │
│      MSSQLQuery,                                                                                                                                                                                                                   
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
ModuleNotFoundError: No module named 'pypika.dialects'

**Reply by yusufix** [reply]
Python 3.12.9 (tags/v3.12.9:fdb8142, Feb  4 2025, 15:27:58) [MSC v.1942 64 bit (AMD64)] on win32
